### ISSUE TITLE
  Accessibility Improvements ‣ Cart ‣ Remove Item 'X' Buttons

### ISSUE THREAD

**Issue by barryhughes** [woothemes/woocommerce#37599]
As originally noted [over here](https://github.com/woocommerce/woocommerce/issues/24185#issuecomment-779227823), there may be room to improve accessibility in relation to the Remove Item ('X') buttons on the cart page.

<img width="809" alt="cart-remove-item" src="https://user-images.githubusercontent.com/3594411/230440250-efcf0928-4c27-4022-b426-b2797cf1c0e5.png">

It already has an `aria-label` attribute, but it seems this may not be sufficient by itself, and it is proposed we add an `aria-describedby` attribute matching a corresponding ID for the product link in the cart table.

**Comment by faisal-alvi** [comment]
Hi @chihsuan 👋🏻 I would like to work on this.

Currently, the product link in the cart table does not have any ID and we must provide its ID in the respective 'X' button's `aria-describedby` attribute.

```
<td class="product-name" data-title="Product">
	<a href="http://localhost:8888/?product=album">Album</a>
</td>
```

What should be added as the ID of each product link in the cart?

![image](https://user-images.githubusercontent.com/25176325/233038780-500be42e-6cee-4b8d-ac5c-3f99ff91127a.png)

**Comment by faisal-alvi** [comment]
Hi @opr 👋🏻  IMO, we should add ID in the format: `id="product-title-XX"`; where XX is the post ID of the product.

### 1. Add `id` to the product link:
```
<td class="product-name" data-title="Product">
	<a id="product-title-14" href="http://localhost:8888/?product=album">Album</a>
</td>
```

### 2. Use the added id in 'X' button's `aria-describedby` attribute:
```
<a href="http://localhost:8888/?page_id=6" class="remove" aria-label="Remove this item" aria-describedby="product-title-14" data-product_id="25" data-product_sku="woo-album">×</a>
```

Please share your thoughts.

**Comment by opr** [comment]
I am wondering if this is a good use of `aria-describedby`, according to [the MDN page on `aria-describedby`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-describedby) the element that is _describing_ the object should offer more detailed information on the object.

In the example on that page, you can see that the element has more detailed information on what it means to "trash" something. This is the example on that page:

```html
<button aria-describedby="trash-desc">Move to trash</button>
…
<p id="trash-desc">
  Items in the trash will be permanently removed after 30 days.
</p>
```

Would simply referencing the product name in the `aria-describedby` attribute offer a good accessibility improvement?

What do you think instead about modifying the `aria-label` to include the product name?

In your example:

```
<a href="http://localhost:8888/?page_id=6" class="remove" aria-label="Remove this item" aria-describedby="product-title-14" data-product_id="25" data-product_sku="woo-album">×</a>
```

what if a product is added to the cart with different attributes, the product ID would be the same but there would be two entries resulting in a duplicated element ID.

e.g. 
<img width="692" alt="image" src="https://user-images.githubusercontent.com/5656702/233045736-52493ef1-a6b9-44d0-adc2-a6e1e90206c5.png">

**Comment by faisal-alvi** [comment]
@opr Adding the product name in the `aria-label` make sense.

Then the aria-label would be: `aria-label="Remove Hoodie - Blue, Yes"`.

Please let me know your thought.

**Comment by opr** [comment]
I think this is better, and easier than adding `aria-describedby`.

I looked into this further I wanted to mention that `aria-describedby` is for "supplemental, non-critical information" (https://benmyers.dev/blog/aria-labels-and-descriptions/). I'd argue the name of the product to be removed _is_ critical.

We should use a placeholder in the translated string to add the product name.
